### PR TITLE
9449 chain facades return vows

### PR DESCRIPTION
**Pull request by turadg**
refs: #9449

## Description
More exos returning vows. Note these are _heap_ vows due to https://github.com/Agoric/agoric-sdk/blob/944adea10660e68a65c10a2b32b6899919315e31/packages/vow/vat.js#L29-L31

### Security Considerations
none

### Scaling Considerations
none

### Documentation Considerations
none

### Testing Considerations
CI

### Upgrade Considerations
none

### CODE REVIEW

File: packages/orchestration/src/exos/remote-chain-facade.js [hunk 1]
```diff
@@ -46,33 +54,38 @@ export const prepareRemoteChainFacade = (
...
-        if (!stakingDenom) {
-          throw Fail`chain info lacks staking denom`;
-        }
-        return makeCosmosOrchestrationAccount(address, stakingDenom, {
-          account: icaAccount,
-          storageNode,
-          icqConnection: anyVal,
-          timer,
+        // FIXME use watch() from vowTools
+        return watch(allVows([icaP, V(icaP).getAddress()]), {
```
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/remote-chain-facade.js:79]: Why not destructure from L42 and wrap the watcher in a `Far`?

**Review by 0xpatrickdev — APPROVED** [review]

File: packages/orchestration/src/exos/remote-chain-facade.js (hunk 1 above)
Comment (turadg) [reply packages/orchestration/src/exos/remote-chain-facade.js:79]: Because I want it to be more likely to fail in testing.

**Comment by dckc** [comment]
> Note these are _heap_ vows due to...

In `provideOrchestration`, we seem to make durable `vowTools`:

```js
   const vowTools = prepareVowTools(zone.subZone('vows'));
```

Do we not pass them around to the relevant places?

File: packages/orchestration/src/exos/remote-chain-facade.js [hunk 2]
```diff
@@ -28,11 +29,18 @@ const anyVal = null;
...
  */
 export const prepareRemoteChainFacade = (
   zone,
-  { makeCosmosOrchestrationAccount, orchestration, storageNode, timer },
+  {
+    makeCosmosOrchestrationAccount,
+    orchestration,
+    storageNode,
+    timer,
+    vowTools: { allVows },
```
Comment (dckc) [inline packages/orchestration/src/exos/remote-chain-facade.js:42]: If we get `watch` from these `vowTools`, it'll make durable vows, right?

File: packages/orchestration/src/exos/remote-chain-facade.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
 /** @file ChainAccount exo */
 import { makeTracer } from '@agoric/internal';
-import { V } from '@agoric/vow/vat.js';
+import { V, watch } from '@agoric/vow/vat.js';
```
Comment (dckc) [inline packages/orchestration/src/exos/remote-chain-facade.js:3]: How about getting `watch` from the durable `vowTools`? See below.

File: packages/orchestration/src/examples/swapExample.contract.js [hunk 4]
```diff
@@ -95,8 +96,10 @@ export const start = async (zcf, privateArgs, baggage) => {
       const agoric = await orch.getChain('agoric');
 
       const [omniAccount, localAccount] = await Promise.all([
-        omni.makeAccount(),
-        agoric.makeAccount(),
+        // XXX when() until membrane
+        when(omni.makeAccount()),
```
Comment (dckc) [inline packages/orchestration/src/examples/swapExample.contract.js:100]: The `vowTools` from the call to `provideOrchestration` are durable, right? How about getting `when` from there?

File: packages/orchestration/src/examples/swapExample.contract.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { withdrawFromSeat } from '@agoric/zoe/src/contractSupport/zoeHelpers.js'
 import { Far } from '@endo/far';
 import { deeplyFulfilled } from '@endo/marshal';
 import { M, objectMap } from '@endo/patterns';
+import { when } from '@agoric/vow/vat.js';
```
Comment (dckc) [inline packages/orchestration/src/examples/swapExample.contract.js:7]: see below re durable vowTools...

**Comment by turadg** [comment]
> Do we not pass them around to the relevant places?

They're available. See https://github.com/Agoric/agoric-sdk/pull/9547/files#r1648321242

File: packages/orchestration/src/exos/local-chain-facade.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /** @file ChainAccount exo */
-import { V } from '@agoric/vow/vat.js';
+import { V, watch } from '@agoric/vow/vat.js';
```
Comment (dckc) [inline packages/orchestration/src/exos/local-chain-facade.js:2]: again, a durable version is reasonably handy...

File: packages/orchestration/src/exos/local-chain-facade.js [hunk 7]
```diff
@@ -21,11 +22,17 @@ import { ChainFacadeI } from '../typeGuards.js';
...
  * }} powers
  */
 export const prepareLocalChainFacade = (
   zone,
-  { makeLocalOrchestrationAccountKit, localchain, storageNode },
+  {
+    makeLocalOrchestrationAccountKit,
+    localchain,
+    storageNode,
+    vowTools: { allVows },
```
Comment (dckc) [inline packages/orchestration/src/exos/local-chain-facade.js:34]: a durable `watch` is available in these `vowTools`, right?

**Comment by turadg** [comment]
@mergifyio requeue

**Review by dckc — COMMENTED** [review]
looks to me like some tiny changes would make the vows durable

**Pull request merged**